packages/oasgraph/README.md

```diff
@@ -1,8 +1,166 @@
+    'x-origin': 'GraphQL' // send header to identify requests made via GraphQL
+  },
+  qs: {
+    limit: 30 // send limit query string in every request
+  },
+  addSubOperations: false
+})
+```
+
+
+
+## Authentication
+Per default, OASGraph will wrap API requests that need authentication in corresponding `viewers`, which allow the user to pass required credentials. OASGraph currently supports viewers for basic authentication and API keys. For example, a query using an API key viewer is:
+
+```javascript
+{
+  viewerApiKey (apiKey: "api_key_here") {
+    ...  // query for authenticated data here
+  }
+}
+```
+
+OASGraph uses dedicated viewers for mutations. For example, a mutation using a basic authentication viewer is:
+
+```javascript
+mutation {
+  mutationViewerBasic (username: "user", password: "secret") {
+    ...  // mutate authenticated data here
+  }
+}
+```
+
+OASGraph further provides `anyAuth` viewers (for queries and mutations), which allow the user to simultaneously provide information for multiple authentication mechanisms. AnyAuth viewers allow OASGraph to resolve nested queries and mutations that encompass API requests with different authentication mechanisms. For example, consider the following query:
+
+```javascript
+{
+  viewerAnyAuth (
+    exampleApiKeyProtocol: {apiKey: "a1p2i3k4e5y"}
+    exampleBasicProtocol: {
+      username: "erik"
+      password: "secret"
+    }
+  ) {
+    patentWithId (patentId: "test") {  // requires "exampleApiKeyProtocol"
+      patentId
+      inventor {                       // requires "exampleBasicProtocol"
+        name
+      }
+    }
+  }
+}
+```
+
+
+## Authorization
+Because OASGraph is a library, it cannot make the callbacks that OAuth requires by itself. Instead, the user must take care of the callback. After the user has obtained the OAuth token from the callback, simply pass the token, specifically the path of the token, to OASGraph through the `tokenJSONpath` [option](./README.md#options).
+
+To see an example of how this would work, click [here](https://github.ibm.com/apiharmony/oasgraph-oauth-github-example)!
+
+
+## Logging
+OASGraph provides multiple levels of logging, which can be controlled by a `DEBUG` environment variable. You can enable these levels using:
+
+```bash
+DEBUG=level_1,level_2 node app-using-oasgraph.js
+```
+
+The following logging levels are supported:
+
+* `preprocessing`: Logs information about preprocessing the OAS to GraphQL.
+* `translation`: Logs information about translating an OAS to GraphQL.
+* `http`: Logs information about the HTTP requests made to the API.
+
+
+## Testing
+To test OASGraph, run:
+
+```bash
+npm test
+```
 
-Turns APIs described by OpenAPI specifications (OAS) into GraphQL interfaces.
+This command will temporarily start and later shut down an example REST(-like) API.
 
-For the full documentation, see the [root of this monorepo](https://github.com/strongloop/oasgraph/).
```

Comment:
I wonder if we should give a reference to the CLI tool and consequently the monorepo...

Reply:
Also the CLI tool still has this comment... and same thing, should the CLI tool README.md also make reference to the library and consequently the monorepo?